### PR TITLE
ENH: Making compatible to Qt5.7 WebEngine

### PR DESCRIPTION
**Pull request by hina-shah**

### CODE REVIEW

**Comment by hina-shah** [comment]
@jcfr : Please have a look!

File: public/js/common/common.js [hunk 1]
```diff
@@ -1,5 +1,13 @@
 var midas = midas || {};
 midas.slicerappstore = midas.slicerappstore || {};
+'use strict';
+
```
Comment (jcfr) [inline public/js/common/common.js]: Add a comment indicating that this is expected to be true if Qt5.x where Web?Engine is used

File: public/js/common/common.js [hunk 2]
```diff
@@ -24,7 +32,7 @@ midas.slicerappstore.extensionButtonClick = function() {
  */
 midas.slicerappstore.updateExtensionButtonState = function(extensionName) {
     if(!window.extensions_manager_model) {
-        midas.slicerappstore.setExtensionButtonState(extensionName, 'Download');
+	midas.slicerappstore.setExtensionButtonState(extensionName, 'Download');
```
Comment (jcfr) [inline public/js/common/common.js]: Is this a change "space" -> "tab" ... ? May it can be removed

File: public/js/common/qwebchannel.js [hunk 3]
```diff
@@ -0,0 +1,427 @@
+/****************************************************************************
+**
+** Copyright (C) 2016 The Qt Company Ltd.
+** Copyright (C) 2016 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com, author Milian Wolff <milian.wolff@kdab.com>
+** Contact: https://www.qt.io/licensing/
+**
+** This file is part of the QtWebChannel module of the Qt Toolkit.
+**
+** $QT_BEGIN_LICENSE:LGPL$
```
Comment (jcfr) [inline public/js/common/qwebchannel.js]: Should we also add mention in the top level LICENSE file of the plugin ?

File: public/js/common/qwebchannel.js [hunk 4]
```diff
@@ -0,0 +1,427 @@
...
+** included in the packaging of this file. Please review the following
+** information to ensure the GNU General Public License requirements will
+** be met: https://www.gnu.org/licenses/gpl-2.0.html and
+** https://www.gnu.org/licenses/gpl-3.0.html.
+**
+** $QT_END_LICENSE$
+**
+****************************************************************************/
+
+"use strict";
```
Comment (jcfr) [inline public/js/common/qwebchannel.js]: Where this file was copied from ? And what the version ? Would be nice to add a comment at the top of the file to explain this.

File: views/extension/view.phtml [hunk 5]
```diff
@@ -16,6 +16,7 @@ $this->headScript()->appendFile($this->coreWebroot.'/public/js/jquery/jqplot/jqp
 $this->headScript()->appendFile($this->coreWebroot.'/public/js/jquery/jqplot/jqplot.pointLabels.min.js');
 $this->headScript()->appendFile($this->coreWebroot.'/public/js/jquery/jquery.autogrow-textarea.js');
 $this->headScript()->appendFile($this->coreWebroot.'/public/js/jquery/jquery.form.js');
+$this->headScript()->appendFile($this->moduleWebroot.'/public/js/common/qwebchannel.js');
```
Comment (jcfr) [inline views/extension/view.phtml]: Would be worth checking that this doesn't have unwanted side effect with the current version of Slicer

File: public/js/extension/extension.view.js [hunk 6]
```diff
@@ -196,3 +195,8 @@ $(document).ready(function() {
         });
     });
 });
+
+$(window).on("load", function() {
+    // Extension button state is updated once all window variables and qt webchannels are set up.
+    midas.slicerappstore.updateExtensionButtonState(json.extension.productname);
```
Comment (jcfr) [inline public/js/extension/extension.view.js]: indent

**Review by jcfr — COMMENTED** [review]
Well done. :+1: 

Few comments to address but overall looks good.

File: public/js/extension/extension.view.js (hunk 6 above)
Comment (hina-shah) [inline public/js/extension/extension.view.js]: Decrease it or increase it?

File: public/js/common/qwebchannel.js (hunk 3 above)
Comment (hina-shah) [inline public/js/common/qwebchannel.js]: Sure.. The file has all the three types of licenses. What should I add?

File: views/extension/view.phtml (hunk 5 above)
Comment (hina-shah) [inline views/extension/view.phtml]: Checked against the current version of Slicer. Does not seem to affect it.

File: public/js/common/qwebchannel.js (hunk 3 above)
Comment (jcfr) [inline public/js/common/qwebchannel.js]: That would be LGPL3

File: public/js/extension/extension.view.js (hunk 6 above)
Comment (jcfr) [inline public/js/extension/extension.view.js]: Sorry for the noise, I got confused by the diff view on GitHub

**Comment by hina-shah** [comment]
@jcfr: Changes made.

File: README [hunk 7]
```diff
@@ -6,3 +6,6 @@ http://www.apache.org/licenses/LICENSE-2.0
 
 The license file was added at revision 19f24fe on 2017-06-07, but you may
 consider that the license applies to all prior revisions as well.
+
+qtwebchannel added after revision 19f24fe on is licensed as LGPL3.
```
Comment (jcfr) [inline README]: `public/js/common/qwebchannel.js` added after revision 19f24fe  and is licensed as LGPL3.

**Comment by jcfr** [comment]
Closing. After verifying that extension could still be installed using Slicer built against Qt4, corresponding changes have been integrated in https://github.com/midasplatform/slicerappstore/commit/838dac3e21cad6f145da3845d78b0e0fd4a7eda4

**Pull request closed without merging**